File: app/src/main/java/lu/aqu/projper/api/ApiModule.java
package lu.aqu.projper.api;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import lu.aqu.projper.api.endpoint.ProjectService;
import retrofit2.Retrofit;

@Module(includes = {RetrofitModule.class})
public class ApiModule {

    @Provides
    @Singleton
    public static ProjectService projectApi(Retrofit retrofit) {
        return retrofit.create(ProjectService.class);
    }

}

File: modules/features/auth/infra/src/main/java/lu/aqu/projper/auth/infra/api/ApiModule.kt
package lu.aqu.projper.auth.infra.api

import dagger.Module
import dagger.Provides
import lu.aqu.projper.auth.infra.localdatasource.AuthDataSource

@Module
class ApiModule {

    @Provides
    fun provideAuthApiClient(): AuthApiClient = AuthDataSource
}
